### ISSUE TITLE
Events stopped being added for an Action, whilst still working for identical duplicate Action

### ISSUE THREAD

**Issue by benjackwhite** [PostHog/posthog#3576]
## Bug description

We have an action defined for a custom event "user signed up" (ID: 32) that filters where a property "test_user = false" and "gender = female". Since one month there are no new events showing in the UI and none of our slack messages are triggered. Today we tried to make a new action (ID: 119) with the same configuration and the events then show correctly.

**DB Query to examine action configuration**
It appears the action match group steps are configured exactly the same:

```
SELECT action_id, event, properties from posthog_actionstep where action_id in (32, 119);

# Old Action (32)
32
"user signed up"
[{"key": "gender", "type": "person", "value": "female", "operator": "exact"}, {"key": "test_user", "type": "person", "value": "false", "operator": "exact"}]

# New Action (119)
119
"user signed up"
[{"key": "gender", "type": "person", "value": "female", "operator": "exact"}, {"key": "test_user", "type": "person", "value": "false", "operator": "exact"}]
```

**DB Query to examine matched events**

```
SELECT action_id, count(*) from posthog_action_events where action_id in (32, 119) GROUP by action_id;
32	2159
119	2383
```

## Expected behavio(u)r

We expect the action to perform reliably in capturing events for both display in dashboards (this is how we noticed - the numbers didn't match our internal database) as well as the slack notifications that we use.

## How to reproduce

Not sure how to get into this state as we can't see any clear difference and when we create a "new" action with the same configuration it seems to work fine. My gut feeling is there might be some cache issues but not sure.

## Environment

- [ ] PostHog self-hosted with Kubernetes 1.9, Helm chart 1.4.8, posthog version: 1.21.0

## Additional context

One month ago when the events stopped for this Action we were on Posthog 1.19.0 I think. We also noticed another Action with the same issue which we "fixed" by deleting and re-adding the filters but this is obviously not ideal. Hence with this action we decided not to modify it so that we can try and investigate what the root cause of the issue is.

**Comment by benjackwhite** [comment]
Worth mentioning, we have Sentry installed and have had no issues reported from the backend related to this sort of issue. Any tips or hints on how to debug further would be really welcome!

**Comment by timgl** [comment]
Hey Ben, thanks for the extensive write up!

- I assume the action wasn't accidentally deleted (ie you can still see it if you go to the url but it doesn't show up in the list)?
- Do you have plugins turned on?
- If you save the action again does it start working?

**Comment by benjackwhite** [comment]
> Hey Ben, thanks for the extensive write up!
> 
> * I assume the action wasn't accidentally deleted (ie you can still see it if you go to the url but it doesn't show up in the list)?
> * Do you have plugins turned on?
> * If you save the action again does it start working?

Hey Tim.
* Action definitely wasn't deleted - it is still in the Actions list as well as by directly visiting it
* Yes we have plugins turned on but are not using them for this project
* It probably will work again (see the additional-context) as we tried this before. The issue is more we can't have the fix simply be "save it and it works again" as then it introduces a general unreliability to the whole use of actions. Does that make sense?

Happy to add any more context or do any deeper looking if you or anyone else on the team has ideas how we could debug further.

**Comment by mariusandra** [comment]
Hey! I think we finally found the issue. We'll release a new version (`1.22.1`) with the fixes shortly, yet to get around this yourself, you can either set the env `PLUGIN_SERVER_INGESTION` to `true` (even if you have manually opted in to plugins)... or alternatively just set `ASYNC_EVENT_ACTION_MAPPING` to `true`. 

This should solve the issue if you're not using webhooks. If you are, then the workaround now is to set both envs to false and disable plugins for your project... or wait until 1.22.1 is out.

**Comment by jaredsilver** [comment]
@mariusandra Do you have a sense of when 1.22.1 might be out?

**Comment by Twixes** [comment]
Hey! Given our release schedule we decided to include the fix with 1.23, which is coming out in a couple of days. `ASYNC_EVENT_ACTION_MAPPING` should work as a solution in the meantime.
